ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java

```diff
@@ -1744,6 +1747,15 @@ public void setForwardedAddresses(List<String> forwardedAddresses) {
   public String getReloadableAuxJars() {
     return StringUtils.join(preReloadableAuxJars, ',');
   }
+
+  public void updateProgressMonitor(ProgressMonitor progressMonitor) {
```

Comment:
lets add synchronize keyword on the get and update ProgressMonitor methods as we will be checking if this is set from CliService.getOperationStatus which will be in a different thread. Alternatively, we can also make this variable volatile.